Rust 1.44 release announcement (#604)
rust-lang/blog.rust-lang.org: posts/2020-06-04-Rust-1.44.0.md

```

If you don't have it already, you can [get `rustup`][install] from the
appropriate page on our website, and check out the [detailed release notes for
1.44.0][notes] on GitHub.

[install]: https://www.rust-lang.org/install.html
[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1440-2020-06-04

## What's in 1.44.0 stable

Rust 1.44 is a smaller release, containing almost exclusively polish and a
few new stabilized standard library APIs.

`cargo tree` is a very old, popular extension to Cargo. Originally released
in December of 2015, and with almost 50,000 downloads, [we have decided to
add it to Cargo directly][cargotree]. If you haven't used it in the past,
`cargo tree` will print your dependency tree to the console:

```text
  mdbook v0.3.2 (/Users/src/rust/mdbook)
├── ammonia v3.0.0
│   ├── html5ever v0.24.0
│   │   ├── log v0.4.8
│   │   │   └── cfg-if v0.1.9
│   │   ├── mac v0.1.1
│   │   └── markup5ever v0.9.0
│   │       ├── log v0.4.8 (*)
│   │       ├── phf v0.7.24
│   │       │   └── phf_shared v0.7.24
│   │       │       ├── siphasher v0.2.3
│   │       │       └── unicase v1.4.2
│   │       │           [build-dependencies]

> │   │       │           └── version_check v0.1.5

Reviewer: Could someone please check that with the CSS used on the blog itself, these line-drawing characters join up properly? GitHub's own CSS breaks line-drawing characters, leaving gaps between them. They *should* look more like this:

![screenshot](https://user-images.githubusercontent.com/162737/83362542-f4e66700-a346-11ea-8eae-4187bce475be.png)

...
```

In this example, the `mdbook` package depends on the `ammonia` package,
which itself depends on `html5ever`, which depends on `log`, `mac`, and
`markup5ever`... and so on.

> 

Reviewer: ```suggestion

You can also use `cargo tree -d` (shorthand for `cargo tree --duplicates`) to show any dependencies on multiple versions of the same crate. This will show one entry for each version of each dependency, and an inverted dependency tree showing how that version gets pulled in.

```

I've avoided giving an example here, to avoid singling out any specific crates (which will likely get fixed in the future).

This release also marks an extension to `async`/`await`: [it can now be used

> in `no_std` context][asyncawaitnostd]. The initial implemenation of this feature required

Reviewer 2: ```suggestion
in `no_std` contexts][asyncawaitnostd]. The initial implemenation of this feature required
```

thread local storage, but that was a temporary situation. In this release,

> this requirement has been removed, meaning that you no longer require
> `libstd` to use `async`/`await`.

Reviewer 3: ```suggestion
this requirement has been removed, meaning that `libstd` is no longer required
to use `async`/`await`.
```

Something feels off about the current phrasing but I don't mind either way.

Reviewer 2: Probably what feels off is that the reader was not "requiring" something, the compiler was. The rephrasing is better; thanks!

[cargotree]: https://github.com/rust-lang/cargo/pull/8062/
[asyncawaitnostd]: https://github.com/rust-lang/rust/pull/69033/

### Library changes

> [`vec![]` can now be used in `const` contexts.][70632] Note that this is only
> empty vectors; you can only do `vec![]`, not `vec![1, 2, 3]`.

Reviewer 2: ```suggestion
[`vec![]` can now be used in `const` contexts.][70632] Note that only
empty vectors can be created; `vec![]` will compile, but `vec![1, 2, 3]` will not.
```

[Unicode 13 is now supported.][69929]

> [`Vec<T>` now implements `From<[T; N]>`.][68692] Where `N` is less than 32.

Reviewer 2: ```suggestion
[`Vec<T>` now implements `From<[T; N]>`][68692] where `N` is less than 32.
```

It might be better to put the `Vec` items next to each other; it feels a bit odd to me to have a completely unrelated item separating them.

add it to Cargo directly][cargotree]. If you haven't used it in the past,
`cargo tree` will print your dependency tree to the console:

```text
  mdbook v0.3.2 (/Users/src/rust/mdbook)
├── ammonia v3.0.0
│   ├── html5ever v0.24.0
│   │   ├── log v0.4.8
│   │   │   └── cfg-if v0.1.9
│   │   ├── mac v0.1.1
│   │   └── markup5ever v0.9.0
│   │       ├── log v0.4.8 (*)
│   │       ├── phf v0.7.24
│   │       │   └── phf_shared v0.7.24
│   │       │       ├── siphasher v0.2.3
│   │       │       └── unicase v1.4.2
│   │       │           [build-dependencies]
│   │       │           └── version_check v0.1.5
...
```

In this example, the `mdbook` package depends on the `ammonia` package,
which itself depends on `html5ever`, which depends on `log`, `mac`, and
`markup5ever`... and so on.

You can also use `cargo tree -d` (shorthand for `cargo tree --duplicates`) to show any dependencies on multiple versions of the same crate. This will show one entry for each version of each dependency, and an inverted dependency tree showing how that version gets pulled in.

This release also marks an extension to `async`/`await`: [it can now be used

> in `no_std` contexts][asyncawaitnostd]. The initial implemenation of this feature required

Reviewer 4: *Did you mean* "implementation"?
